modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/disruptor/delivery/DisruptorBasedFlusher.java

```diff
@@ -105,13 +107,27 @@ public DisruptorBasedFlusher() {
             contentReadTaskBatchProcessor[i].setExceptionHandler(new DeliveryExceptionHandler());
         }
 
+        // Initialize decompression handlers
+        DisruptorBasedContentDecompressionHandler[] decompressionEventHandlers =
+                new DisruptorBasedContentDecompressionHandler[parallelDecompressionHandlers];
+        if (ContentCompressionDecompressionHandler.isCompressionEnabled) {
```

Comment:
usually directly accessing variables is not recommended. is this from a configuration parameter?